### PR TITLE
Add Scalar DB supported databases

### PR DESCRIPTION
**Pull request by scalar-boney**
Add Scalar DB compatibility matrix to docs

Issues:
*  Scalar DB 2.2.2 and 2.3.0 with Cosmos DB does not support partial update with `storage` mode
* We should manage `com.codahale.metrics.JmxReporter` dependency while using Scalar DB 2.1.0 and 2.0.1 with spring boot application.
* DynamoDB schema creation failed with `scalar-schema-standalone-2.3.0.jar` due to Cannot cast java.lang.Long to java.lang.Integer
* Cassandra 2.2.19 schema deletion failed with  scalar-schema-standalone-2.2.2.jar

### CODE REVIEW

**Comment by brfrn169** [comment]
@scalar-boney The issues in your first comment mean bugs we should fix?

**Comment by scalar-boney** [comment]
> @scalar-boney The issues in your first comment mean bugs we should fix?

Sorry, I am not sure.
It is a known issue.
I think we can fix this issue with `get ' before the update (I mean not a partial update).

**Comment by feeblefakie** [comment]
@brfrn169 @scalar-boney 
Maybe, the old Cosmos DB adapter doesn't use a stored procedure for a simple put so it doesn't upsert properly.

File: docs/compatibility_matrix.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Compatibility Matrix
+
+This document shows the compatibility of Scalar DB with supported databases.
+
+## Scalar DB Compatibility
+
+| Scalar DB Version   | Cassandra 3.11.10 | Cassandra 3.0.24    | Cassandra 2.2.19   | Cosmos DB | DynamoDB  | MySQL 8.0.25  | MySQL 2.7.34  | PostgreSQL 13.3    | PostgreSQL 12.7   | PostgreSQL 11.12  |
```
Comment (feeblefakie) [inline docs/compatibility_matrix.md:7]: MySQL 2.7 sounds very old.
Is it a mistake? and 5.7?

Comment (feeblefakie) [inline docs/compatibility_matrix.md:7]: I'm not sure if we should note the bug-fix versions for Cassandra and MySQL. Should they be `3.11` and `8.0` respectvively?

Comment (scalar-boney) [reply docs/compatibility_matrix.md:7]: Oh, sorry!
It's my mistake, 5.7 is correct.

Comment (scalar-boney) [reply docs/compatibility_matrix.md]: Cassandra and MySQL compatibility updated with stable versions
MySQL
```
$ brew info mysql@8.0
mysql: stable 8.0.25 (bottled)

$ brew info mysql@5.7
mysql@5.7: stable 5.7.34 (bottled) 
```
Screen Shot 2021-07-27 at 12 41 00 PM

Comment (scalar-boney) [reply docs/compatibility_matrix.md]: Thank you for your review comments

File: docs/compatibility_matrix.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Compatibility Matrix
+
+This document shows the compatibility of Scalar DB with supported databases.
+
+## Scalar DB Compatibility
+
+| Scalar DB Version   | Cassandra 3.11 | Cassandra 3.0    | Cassandra 2.2   | Cosmos DB | DynamoDB  | MySQL 8.0  | MySQL 5.7  | PostgreSQL 13.3    | PostgreSQL 12.7   | PostgreSQL 11.12  |
+|:--------------------|:------------------|:--------------------|:-------------------|:----------|:----------|:--------------|:--------------|:-------------------|:------------------|:------------------|
+| Scalar DB 3.0.0 | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ |
```
Comment (Torch3333) [inline docs/compatibility_matrix.md:9]: Do you think the table would be easier to grasp in a glance if ✓ and x were replaced by emoji such as ✅ and ❌ ?

**Review by Torch3333 — COMMENTED** [review]
Thank you for computing these results !! I left one comment regarding the style.

File: docs/compatibility_matrix.md (hunk 2 above)
Comment (scalar-boney) [reply docs/compatibility_matrix.md]: Thank you for your suggestion.
Document updated with emoji.

File: docs/compatibility_matrix.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Compatibility Matrix
```
Comment (feeblefakie) [inline docs/compatibility_matrix.md:1]:
```suggestion
# Scalar DB Supported Databases
```

File: docs/compatibility_matrix.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+This document shows the compatibility of Scalar DB with supported databases.
+
+## Scalar DB Compatibility
```
Comment (feeblefakie) [inline docs/compatibility_matrix.md:5]:
```suggestion
## Compatibility matrix
```

File: docs/compatibility_matrix.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Compatibility Matrix
+
+This document shows the compatibility of Scalar DB with supported databases.
```
Comment (feeblefakie) [inline docs/compatibility_matrix.md:3]:
```suggestion
Scalar DB supports the following databases and the databases that are compatible with those databases.
* aaa
* bbb 
```

**Review by feeblefakie — COMMENTED** [review]
Why don't you make this document more like a `Scalar DB supported databases`?
We don't really mention the supported databases in the docs so we need something like this.
I'm not fully sure if they are in the same doc but I feel why not.

File: docs/compatibility_matrix.md (hunk 3 above)
Comment (scalar-boney) [reply docs/compatibility_matrix.md]: Thank you for your suggestion.
Suggestion integrated.

File: docs/compatibility_matrix.md (hunk 4 above)
Comment (scalar-boney) [reply docs/compatibility_matrix.md]: Thank you for your suggestion.
Suggestion integrated.

File: docs/compatibility_matrix.md (hunk 5 above)
Comment (scalar-boney) [reply docs/compatibility_matrix.md]: Thank you for your suggestion.
Suggestion integrated.

**Review by feeblefakie — DISMISSED** [review]
LGTM! Thanks.
Can you add the link from the top-level README?

Also, please work on Oracle in another PR.

**Comment by scalar-boney** [comment]
> LGTM! Thanks.
> Can you add the link from the top-level README?

Sorry for the delay.
Compatibility matrix link added to  top-level README 024b3d5.

> Also, please work on Oracle in another PR.

Yes, I will create it later.

File: README.md [hunk 6]
```diff
@@ -20,6 +20,7 @@ dependencies {
     * [Making Cassandra more capable, faster, and more reliable](https://www.slideshare.net/scalar-inc/making-cassandra-more-capable-faster-and-more-reliable-at-apacheconhome-2020) at ApacheCon@Home 2020
     * [Scalar DB: A library that makes non-ACID databases ACID-compliant](https://www.slideshare.net/scalar-inc/scalar-db-a-library-that-makes-nonacid-databases-acidcompliant) at Database Lounge Tokyo #6 2020
     * [Transaction Management on Cassandra](https://www.slideshare.net/scalar-inc/transaction-management-on-cassandra) at Next Generation Cassandra Conference / ApacheCon NA 2019
+* [Scalar DB compatibility martix](docs/compatibility_matrix.md)
```
Comment (feeblefakie) [inline README.md:23]:
```suggestion
* [Scalar DB supported databases](docs/compatibility_matrix.md)
```
Please move it under `Getting started`. Please also change the file name as well.

**Review by feeblefakie — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (scalar-boney) [reply README.md]: Thank you for your suggestion.
Document position and file name also changed 2110824ef

**Review by feeblefakie — APPROVED** [review]
LGTM! Thanks!

So, like I said, Scalar DB 3.1 supports Oracle now, so let's add it in another PR.

**Pull request merged**